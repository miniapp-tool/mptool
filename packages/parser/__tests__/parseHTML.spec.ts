import { expect, it } from "vitest";

import { parseHTML } from "../src/index.js";

it("parseHTML()", () => {
  const cases = [
    // snippet
    '<div class="test">hello</div>',
    '<div class="test"><span>hello</span></div>',
    '<div class="test"><span>hello</span><span>world</span></div>',
    "<table><tr><td>hello</td></tr></table>",
    "<table><tr><td>hello</td><td>world</td></tr></table>",

    // full document
    "<!doctype html><html><head><title>hello</title></head><body>world</body></html>",

    // svg
    "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'><circle cx='50' cy='50' r='40' fill='red'/></svg>",
  ];

  cases.forEach((content) => {
    expect(parseHTML(content)).toMatchSnapshot();
  });
});
