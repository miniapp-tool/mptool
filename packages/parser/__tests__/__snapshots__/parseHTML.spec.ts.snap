// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseHTML() 1`] = `
[
  <div
    class="test"
  >
    hello
  </div>,
]
`;

exports[`parseHTML() 2`] = `
[
  <div
    class="test"
  >
    <span>
      hello
    </span>
  </div>,
]
`;

exports[`parseHTML() 3`] = `
[
  <div
    class="test"
  >
    <span>
      hello
    </span>
    <span>
      world
    </span>
  </div>,
]
`;

exports[`parseHTML() 4`] = `
[
  <table>
    <tr>
      <td>
        hello
      </td>
    </tr>
  </table>,
]
`;

exports[`parseHTML() 5`] = `
[
  <table>
    <tr>
      <td>
        hello
      </td>
      <td>
        world
      </td>
    </tr>
  </table>,
]
`;

exports[`parseHTML() 6`] = `
[
  ProcessingInstruction {
    "data": "!doctype html",
    "endIndex": null,
    "name": "!doctype",
    "next": <html>
      <head>
        <title>
          hello
        </title>
      </head>
      <body>
        world
      </body>
    </html>,
    "parent": Document {
      "children": [
        [Circular],
        <html>
          <head>
            <title>
              hello
            </title>
          </head>
          <body>
            world
          </body>
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
    },
    "prev": null,
    "startIndex": null,
    "type": "directive",
  },
  <html>
    <head>
      <title>
        hello
      </title>
    </head>
    <body>
      world
    </body>
  </html>,
]
`;

exports[`parseHTML() 7`] = `
[
  <svg
    viewbox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="50"
      cy="50"
      fill="red"
      r="40"
    />
  </svg>,
]
`;
